pretyping/find_subterm.ml

```diff
@@ -86,6 +86,7 @@ let replace_term_occ_gen_modulo env sigma occs like_first test bywhat cl occ t =
   let rec substrec k t =
     if Locusops.occurrences_done !count then t else
     try
+      (if not (EConstr.Vars.noccur_between sigma 1 k t) then raise (NotUnifiable None));
```

Comment:
this is probably too violent but let's see if CI can tell

Reply:
The test suite complexity/pattern example started timing out, let's try putting the check in unification instead.